pkg/bundle/bundle.go

```diff
@@ -67,55 +67,85 @@ type Bundle struct {
 }
 
 // ReadConfig takes a bundle spec file as a string and maps it into the Bundle struct
-func ReadConfig(file string) (*BundleConfig, error) {
-	var body []byte
+func ReadConfig(file []string) (*BundleConfig, error) {
+	var tempBundleConfig struct {
+		Addons []*Bundle `yaml:"addons"`
+	}
+	bundleconfig := &BundleConfig{}
 
-	if file == "" {
-		defaultFile, err := getDefaultBundle()
-		if err != nil {
-			return nil, fmt.Errorf("failed retrieving default bundle file: %v", err)
-		}
-		body, err = io.ReadAll(defaultFile)
+	if len(file) == 0 {
+		files, err := defaultBundle.ReadDir("bundles")
 		if err != nil {
-			return nil, fmt.Errorf("unable to read default bundle file: %v", err)
+			fmt.Errorf("unable to process bundles: %v", err)
 		}
-	} else {
-		var err error
-		body, err = ioutil.ReadFile(file)
-		if err != nil {
-			return nil, fmt.Errorf("unable to read user provided file: %v", err)
-		}
-	}
 
-	if len(body) < 1 {
-		return nil, fmt.Errorf("file is empty")
-	}
+		for _, file := range files {
+			f, err := defaultBundle.ReadFile(filepath.Join("bundles", file.Name()))
+			if err != nil {
+				fmt.Printf("unable to read file: %v", err)
+				continue
+			}
 
-	bundleconfig := &BundleConfig{}
-	err := yaml.Unmarshal(body, bundleconfig)
-	if err != nil {
-		return nil, fmt.Errorf("unable to parse yaml file: %v", err)
-	}
-	return bundleconfig, nil
-}
+			if err := yaml.Unmarshal(f, &tempBundleConfig); err != nil {
+				fmt.Printf("Error unmarshaling file %s: %v\n", f, err)
+				continue
+			}
 
-func getDefaultBundle() (fs.File, error) {
-	var filename string
-	filenames, err := fs.ReadDir(defaultBundle, "bundles")
-	if err != nil {
-		return nil, err
-	}
-// TODO: This just grabs the last bundle file in the dir, need to compile them all together to support multiple
-	for _, f := range filenames {
-		if !f.IsDir() {
-			filename = "bundles/" + f.Name()
+			bundleconfig.Addons = append(bundleconfig.Addons, tempBundleConfig.Addons...)
+			value := *bundleconfig
+			fmt.Println(value)
 
 		}
+		// if err != nil {
+		// 	return nil, fmt.Errorf("failed retrieving default bundle files: %v", err)
+		// }
+		// body = append(body, defaultFiles...)
+		// err = yaml.Unmarshal(body, &bundleconfig)
 	}
 
-	file, err := defaultBundle.Open(filename)
-	if err != nil {
-		return nil, err
+	if len(file) > 0 {
+		for _, str := range file {
```

Comment:
I feel like this loop is repeated above. Can we pull it out into a dedicated function?

Reply:
We discussed this in person, but for transparency:

The two loops are due to the fact that they are acting on two different data types that use different read methods. Can't think of a way to make that more generic. We did change it to an if/else instead of 2 ifs to make it a little easier to read.